### PR TITLE
Dp 26581 document mayflower dts

### PR DESCRIPTION
**Pull request by clairesunstudio**

### CODE REVIEW

File: packages/react/README.md [hunk 1]
```diff
@@ -55,6 +67,40 @@ import Button from '@massds/mayflower-react/dist/Button';
...
 
+### To consume the typed components in your typescript project:
+1. Create a `mayflower.d.ts` in your project and declare module imports by mayflower-react:
+```javascript
+declare module "@massds/mayflower-assets";
+declare module "@massds/mayflower-assets/static/images/*";
+```
+2. Import a mayflower-react typed component. For examples:
+```javascript
+import BrandBanner from "@massds/mayflower-react/dist/BrandBanner";
```
Comment (clairesunstudio) [inline packages/react/README.md]: how does the project know which index file to import? does the entry point need to be specified somewhere?

File: packages/react/README.md [hunk 2]
```diff
@@ -34,15 +34,30 @@ As of version 10.x, Mayflower React styles come directly from the `@massds/mayfl
  */
 ```
 
+As of version 12.x, mayflower-react includes typed components! The typescript version of the components are converted from the source code `src/index.js`. This is done by [transforming proptypes to typescript types using `jscodeshift`](./scripts/jsx-to-tsx), with some custom modifications, see [packages/react/scripts/transform.ts](./scripts/transform.ts).
```
Comment (DmitriiTartynovLCM) [inline packages/react/README.md]: We only have TS declarations, components are still written in JS. And, the `jsx-to-tsx` script is not involved in the generation at the moment.

```suggestion
As of version 12.x, mayflower-react includes type declarations for all the components! These typescript declarations are converted from the same source code `src/index.js`. This is done by transforming proptypes to typescript types using `jscodeshift`, with some custom modifications, see [packages/react/scripts/transform.ts](./scripts/transform.ts).
```

File: packages/react/README.md [hunk 3]
```diff
@@ -34,15 +34,30 @@ As of version 10.x, Mayflower React styles come directly from the `@massds/mayfl
...
+
+Alongside the `index.js` and `index.mjs` (ES5 and ES6 versions), inside of each components in `dist/`, you will find a `index.d.ts` (type declarations) generated during the `rush build:react` step. 
+> You can manually run the convertion script on a specific component. For example, run this in the repo root:
+
+> $ `node packages/react/scripts/jsx-to-tsx packages/react/src/components/atoms/buttons/Button/index.js`
+
+> This command will create create a new `.tsx` copy of the Button index.js file and converts it to typescript, in the same place. 
+
+> This is an intermediate step of the final distribution. In the build step, this script will run on each component, then the generated type declarations of the component will be extracted into a `d.ts` file and placed in the distribution folder.
```
Comment (DmitriiTartynovLCM) [inline packages/react/README.md]:
```suggestion
> This replicates an intermediate step of the final distribution. In the build step, a similar conversion process will run on each component saving its results into the `types` folder, then the generated type declarations of the component will be extracted into a `d.ts` file and placed in the distribution folder. 
```

File: packages/react/README.md [hunk 4]
```diff
@@ -51,9 +66,48 @@ import Button from '@massds/mayflower-react/dist/Button';
 ```javascript
 <Button text="Button" onClick={() => console.log('mayflower button clicked!')} />
 ```
+>For a more detailed guide and information on the components included in Mayflower React and their functionality, visit our [Mayflower React Storybook][react-storybook]. Click on the Info and Knobs tabs for component prop types, details and options.
+
+4. Component styles are imported separated, follow the scss modules documentation in each component. Create a scss file and import the necessary styles from mayflower-assets for the React component in use. E.g. If you using the slim header and slim footer components, in a SCSS files, import these styles:
```
Comment (DmitriiTartynovLCM) [inline packages/react/README.md]:
```suggestion
4. Component styles must be imported separately, follow the scss modules documentation in each component. Create a scss file and import the necessary styles from mayflower-assets for the React component in use. E.g. If you using the slim header and slim footer components, in a SCSS files, import these styles:
```

File: packages/react/README.md (hunk 1 above)
Comment (DmitriiTartynovLCM) [reply packages/react/README.md]: Could we say that it's just `@massds/mayflower-react/dist/[ComponentName]` with the `[ComponentName]` taken from the React storybook?
Selection_012

File: packages/react/README.md [hunk 5]
```diff
@@ -51,9 +66,48 @@ import Button from '@massds/mayflower-react/dist/Button';
...
+@use "~@massds/mayflower-assets/scss/01-atoms/site-logo";
+@use "~@massds/mayflower-assets/scss/03-organisms/footer-slim";
+```
+
+5. Config SCSS import paths
+When using this mayflower-assets for the .scss files, you will need to include its include paths, see [mayflower-assets](packages/assets/README.md) for more info.
+
+
+### To consume the typed components in your typescript project
+1. Create a `mayflower.d.ts` in your project and declare module imports by mayflower-react:
```
Comment (DmitriiTartynovLCM) [inline packages/react/README.md]: This is actually an optional step. It's only necessary if the project is directly importing something from `@massds/mayflower-assets` into a TS code. It turned out we could even remove it from Data Hub, but let's keep it for now.

File: packages/react/README.md [hunk 6]
```diff
@@ -51,9 +66,48 @@ import Button from '@massds/mayflower-react/dist/Button';
...
+declare module "@massds/mayflower-assets";
+declare module "@massds/mayflower-assets/static/images/*";
+```
+2. Import a mayflower-react typed component. For examples:
+```javascript
+import BrandBanner from "@massds/mayflower-react/dist/BrandBanner";
+import HeaderSlim from "@massds/mayflower-react/dist/HeaderSlim";
+import FooterSlim from "@massds/mayflower-react/dist/FooterSlim";
+```
+3. Component styles are imported separated, follow the scss modules documentation in each component. Create a scss file and import the necessary styles from mayflower-assets for React component in use.
```
Comment (DmitriiTartynovLCM) [inline packages/react/README.md]:
```suggestion
3. Component styles must be imported separately, follow the scss modules documentation in each component. Create a scss file and import the necessary styles from mayflower-assets for React component in use. 
```

**Pull request merged**